manifests/params.pp

```diff
@@ -23,121 +22,269 @@
+    'offsets.retention.check.interval.ms'           => '600000',
+    'offsets.retention.minutes'                     => '1440',
+    'offsets.topic.compression.codec'               => '0',
+    'offsets.topic.num.partitions'                  => '50',
+    'offsets.topic.replication.factor'              => '3',
+    'offsets.topic.segment.bytes'                   => '104857600',
+    'port'                                          => '9092',
+    'queued.max.requests'                           => '500',
+    'quota.consumer.default'                        => '9223372036854775807',
+    'quota.producer.default'                        => '9223372036854775807',
     'replica.fetch.max.bytes'                       => '1048576',
-    'replica.fetch.wait.max.ms'                     => '500',
     'replica.fetch.min.bytes'                       => '1',
-    'num.replica.fetchers'                          => '1',
+    'replica.fetch.wait.max.ms'                     => '500',
     'replica.high.watermark.checkpoint.interval.ms' => '5000',
-    'fetch.purgatory.purge.interval.requests'       => '10000',
-    'producer.purgatory.purge.interval.requests'    => '10000',
+    'replica.lag.time.max.ms'                       => '10000',
+    'replica.socket.receive.buffer.bytes'           => '65536',
+    'replica.socket.timeout.ms'                     => '30000',
+    'request.timeout.ms'                            => '30000',
+    'socket.receive.buffer.bytes'                   => '102400',
+    'socket.request.max.bytes'                      => '104857600',
+    'socket.send.buffer.bytes'                      => '102400',
+    'unclean.leader.election.enable'                => true,
+    'zookeeper.connection.timeout.ms'               => '',
     'zookeeper.session.timeout.ms'                  => '6000',
-    'zookeeper.connection.timeout.ms'               => '6000',
-    'zookeeper.sync.time.ms'                        => '2000',
+    'zookeeper.set.acl'                             => false,
+    'broker.id.generation.enable'                   => true,
+    'connections.max.idle.ms'                       => '600000',
     'controlled.shutdown.enable'                    => true,
     'controlled.shutdown.max.retries'               => '3',
     'controlled.shutdown.retry.backoff.ms'          => '5000',
-    'auto.leader.rebalance.enable'                  => true,
-    'leader.imbalance.per.broker.percentage'        => '10',
-    'leader.imbalance.check.interval.seconds'       => '300',
-    'offset.metadata.max.bytes'                     => '1024',
+    'controller.socket.timeout.ms'                  => '30000',
+    'default.replication.factor'                    => '1',
+    'fetch.purgatory.purge.interval.requests'       => '1000',
+    'group.max.session.timeout.ms'                  => '30000',
+    'group.min.session.timeout.ms'                  => '6000',
+    'inter.broker.protocol.version'                 => '0.8.2.2',
```

Comment:
http://kafka.apache.org/documentation.html#upgrade

You can not set this to the current versio,  until the whole cluster has been upgraded?